mongodbatlas/resource_mongodbatlas_privatelink_endpoint_service_adl.go

```diff
@@ -0,0 +1,189 @@
+package mongodbatlas
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	errorADLServiceEndpointAdd = "error adding MongoDB ADL Private Link Endpoint Connection(%s): %s"
+)
+
+func resourceMongoDBAtlasPrivateLinkEndpointServiceADL() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceMongoDBAtlasPrivateLinkEndpointServiceADLCreate,
+		ReadContext:   resourceMongoDBAtlasPrivateLinkEndpointServiceADLRead,
+		UpdateContext: resourceMongoDBAtlasPrivateLinkEndpointServiceADLUpdate,
+		DeleteContext: resourceMongoDBAtlasPrivateLinkEndpointServiceADLDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMongoDBAtlasPrivateLinkEndpointServiceADLImportState,
+		},
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"endpoint_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"provider_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"AWS"}, false),
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"DATA_LAKE"}, false),
+			},
+			"comment": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceMongoDBAtlasPrivateLinkEndpointServiceADLUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// Get client connection.
+	conn := meta.(*MongoDBClient).Atlas
+	ids := decodeStateID(d.Id())
+	projectID := ids["project_id"]
+	endpointID := ids["endpoint_id"]
+
+	privateLink, _, err := conn.DataLakes.GetPrivateLinkEndpoint(ctx, projectID, endpointID)
+	if err != nil {
+		return diag.Errorf("error getting private link endpoint information: %s", err)
```

Comment:
Shouldn't these call out specifically ADL like others?
```suggestion
		return diag.Errorf("Error getting ADL PrivateLink Endpoint Information: %s", err)
```

Reply:
done!